### PR TITLE
#193 Mongodb and python containers security and user auth

### PR DESCRIPTION
**Pull request by guttenberger**

### CODE REVIEW

File: production.docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+# Keeps the MongoDB data consistent between shutdowns
+volumes:
+  mongodb_data_container:
+
+# Create network for internal services
+networks:
+  internal-network:
+    driver: bridge
+    internal: true
```
Comment (Fr3574) [inline production.docker-compose.yml:76]:
```suggestion
    internal: true
    
```

File: .env [hunk 2]
```diff
@@ -17,4 +17,11 @@ PYTHON_BACKEND_PORT=50000
...
 # MongoDB
 MONGODB_PORT=27017
+
+# Only used in development
 MONGODB_URL=mongodb://mongodb:${MONGODB_PORT}/datastore
+
+# Only used in production
+PROD_MONGODB_USERNAME=admin
+PROD_MONGODB_PASSWORD=REPLAC3_ME
+PROD_MONGODB_URL=mongodb://${PROD_MONGODB_USERNAME}:${PROD_MONGODB_PASSWORD}@mongodb:${MONGODB_PORT}
```
Comment (Fr3574) [inline .env:27]:
```suggestion
PROD_MONGODB_URL=mongodb://${PROD_MONGODB_USERNAME}:${PROD_MONGODB_PASSWORD}@mongodb:${MONGODB_PORT}

```

**Comment by guttenberger** [comment]
@Fr3574 can you test the new deployment

**Comment by Fr3574** [comment]
I had to change the nginx port to `8080` (see [Docs](https://github.com/nginxinc/docker-nginx-unprivileged)), otherwise the deployment and docker-compose in general won't work.

**Review by Fr3574 — APPROVED** [review]

**Review by Corgam — APPROVED** [review]

**Pull request merged**